### PR TITLE
Shortcut: Covert component to TypeScript

### PR DESCRIPTION
**Pull request by walbo**
## What?
Converts the `Shortcut` component to TypeScript.

## Why?
Part of the @wordpress/components's TypeScript migration (https://github.com/WordPress/gutenberg/issues/35744).

## How?
- Refactors the current `Shortcut` to TypeScript.
- Refactors the test to TypeScript and refactored to use `@testing-library/react`.

## Testing Instructions
1. Run npm run dev and ensure no typing errors
2. Test within the Gutenberg editor

### CODE REVIEW

File: packages/components/src/shortcut/test/index.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+		render(
+			<Shortcut
+				shortcut={ {
+					display: 'shortcut text',
+					ariaLabel: 'shortcut label',
+				} }
+			/>
+		);
+
+		expect( screen.getByText( 'shortcut text' ) ).toMatchSnapshot();
```
Comment (ciampo) [inline packages/components/src/shortcut/test/index.tsx]: I wonder if we could write this test in a way that explicitly tests the accessible label, maybe something like:

```suggestion
		expect( screen. getByLabelText( 'shortcut label' ) ).toHaveTextContent('shortcut text');
```

**Review by ciampo — APPROVED** [review]
LGTM 🚀

File: packages/components/src/shortcut/test/index.tsx (hunk 1 above)
Comment (walbo) [reply packages/components/src/shortcut/test/index.tsx]: Yep. Will take a look at it 👍

Comment (ciampo) [reply packages/components/src/shortcut/test/index.tsx]: https://github.com/WordPress/gutenberg/pull/42272/commits/041ee3b952f4a6be92ecd3cae8b90ae142e6de40 looks good!

In general, my preference would be to prefer this style of tests over snapshots when possible.

Thank you, and feel free to merge!

**Pull request merged**